fmt_xlong.c

```diff
@@ -0,0 +1,15 @@
+#include "fmt.h"
+
+/* writes ulong u in hex to char *s, does not NULL-terminate */
+unsigned int fmt_xlong(char *s, unsigned long u)
+{
+ unsigned int len; unsigned long q; unsigned long c;
```

Comment:
As I was forced to unwrap the gen_allocdefs code recently: please put only one instruction per line, and only one type of variable.

Reply:
Done.